### ISSUE TITLE
CherryTree hangs on increasing width of Code Box

### ISSUE THREAD

**Issue by Anon-Exploiter** [giuspen/cherrytree#991]
Hi, 

This issue has been recently occurring with me, Cherrytree hangs whenever I add the following code snippet below in the Codebox with format `sh` it hangs. 

```sh
POST /batch HTTP/1.1
Host: 127.0.0.1:8000
User-Agent: Mozilla/5.0 (X11; Linux x86_64; rv:68.0) Gecko/20100101 Firefox/68.0
Accept: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8
Accept-Language: en-US,en;q=0.5
Accept-Encoding: gzip, deflate
Connection: close
Upgrade-Insecure-Requests: 1
Content-Type: application/json
Content-Length: 232

{ "requests": [
            { "method": "get", "path": "/profile" }, { "method": "get", "path": "/item" }, { "method": "get", "path": "/test" }, { "method": "get", "path": "/item/$1.id;require('util').log('s');" }
        ]
    }
```

Cherry Tree Version: **CherryTree 0.39.4** (latest)

Saving format: `.ctb sqlite`

Here's a **video POC** of the issue for better reproduction: https://drive.google.com/file/d/1F0Ml_tsWNkXrQF8ZVtL0ceoDRynJDO5i/view?usp=sharing

Distribution:
```sh
$ cat /etc/*release
PRETTY_NAME="Kali GNU/Linux Rolling"
NAME="Kali GNU/Linux"
ID=kali
VERSION="2020.2"
VERSION_ID="2020.2"
VERSION_CODENAME="kali-rolling"
ID_LIKE=debian
ANSI_COLOR="1;31"
HOME_URL="https://www.kali.org/"
SUPPORT_URL="https://forums.kali.org/"
BUG_REPORT_URL="https://bugs.kali.org/"
```

Last logs which were seen in the console before hanging:
![image](https://user-images.githubusercontent.com/18597330/86642792-9ff8d900-bff5-11ea-8cb7-20f79ea1a08c.png)

Also, the file is being saved in a symlinked directory shared between VM and host machine.

**Comment by txe** [comment]
@giuspen , can you reproduce this issue? I tried but everything was ok.

**Comment by Anon-Exploiter** [comment]
@txe do let me know If I can help you guys with any logs, I'm not sure of any argument which can bring detailed logs. 

If there's any, let me know.

**Comment by txe** [comment]
Maybe it is an issue with the source.  Did you try to copy the code snipped from github? Another thing you can try is to save a document then open it and change codebox width.

**Comment by giuspen** [comment]
@txe I'll try and let you know, mind it's on the old Pygtk2 version. Anon would you be happy to test the latest development version 0.99.4 that is built on much newer library? (I suspect a gtksourceview issue)

**Comment by Anon-Exploiter** [comment]
Thanks, guys. Let me try, will update accordingly.

**Comment by Anon-Exploiter** [comment]
So guys, I'm not sure why, but the issue isn't reproducible. I did it yesterday like 3-4 times and it was hanging but it seems it's fixed somehow. Not sure how, but whatever works :sweat_smile: 

Thanks. Closing this. 

> @txe I'll try and let you know, mind it's on the old Pygtk2 version. @Anon-Exploiter would you be happy to test the latest development version 0.99.4 that is built on much newer library? (I suspect a gtksourceview issue)

I'm on Ubuntu 18.04, that one is for Ubuntu 20.04, tried building but it isn't working, tried resolving the dependency issues which resulted in generating even more errors.
